olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -107,17 +107,17 @@ read::QuadTreeIndex FindQuadTree(
       std::min<std::uint32_t>(tile_key.Level(), kMaxQuadTreeIndexDepth);
   for (int i = max_depth; i >= 0; --i) {
     const auto& root_tile_key = tile_key.ChangedLevelBy(-i);
-    auto cached_tree =
-        repository.Get(layer, root_tile_key, kAggregateQuadTreeDepth, version);
-    if (!cached_tree.IsNull()) {
-      OLP_SDK_LOG_DEBUG_F(kLogTag,
-                          "FindQuadTreeInCache found in cache, tile='%s', "
-                          "root='%s', depth='%" PRId32 "'",
-                          tile_key.ToHereTile().c_str(),
-                          root_tile_key.ToHereTile().c_str(),
-                          kAggregateQuadTreeDepth);
-
-      return cached_tree;
+    for (std::uint32_t depth = i; depth <= max_depth; depth++) {
+      auto cached_tree = repository.Get(layer, root_tile_key, depth, version);
+      if (!cached_tree.IsNull()) {
+        OLP_SDK_LOG_DEBUG_F(kLogTag,
+                            "FindQuadTreeInCache found in cache, tile='%s', "
+                            "root='%s', depth='%" PRId32 "'",
+                            tile_key.ToHereTile().c_str(),
+                            root_tile_key.ToHereTile().c_str(), depth);
+
+        return cached_tree;
+      }
```

Comment:
This will lead to many extra cache searches if tile metadata is not present in the cache.
Maybe we can use always quad tree with depth 4 for GetTile, GetAggregatedData and Prefetch?